program/src/state.rs

```diff
@@ -705,6 +705,9 @@ impl Validator {
     pub fn has_stake_accounts(&self) -> bool {
         self.stake_seeds.begin != self.stake_seeds.end
     }
+    pub fn has_unstake_accounts(&self) -> bool {
+        self.unstake_seeds.begin != self.unstake_seeds.end
+    }
```

Comment:
We need to update `check_can_be_removed` below to also refuse removal if the validator has unstake accounts!